core/src/main/scala/chisel3/ModuleImpl.scala

```diff
@@ -241,10 +241,6 @@ private[chisel3] trait ObjectModuleImpl {
     /** Explicitly Asynchronous Reset */
     case object Asynchronous extends Type
   }
-
-  def getModulePrefixList: List[String] = {
-    Builder.getModulePrefixList
-  }
```

Comment:
I removed this API because it's complicated by the separator being optional and it isn't currently used (it was used internally in `Instantiate` but its use would technically be wrong with separator now being optional).